### DISCUSSION TITLE
Support for RabbitMQ streams

### DISCUSSION THREAD

**Discussion by meirmedigate** [Lancetnik/Propan#100] [Ideas]
Streams are supported in RabbitMQ since 3.9.0.
It would be nice if there would be support for this as well.

**Comment by Lancetnik** [comment]
Well, It is a good idea. Wait for it in days.

**Comment by Lancetnik** [comment]
@meirmedigate Is that are you looking for?

```python
from propan import PropanApp, RabbitBroker
from propan.brokers.rabbit import RabbitQueue
from propan.annotations import Logger

broker = RabbitBroker(consumers=10)
app = PropanApp(broker)

queue = RabbitQueue(
    name="test",
    durable=True,
    arguments={
        "x-queue-type": "stream",
    },
)

@broker.handle(queue, consume_arguments={"x-stream-offset": "first"})
async def handle(msg, logger: Logger):
    logger.info(msg)

@app.after_startup
async def test():
    await broker.publish("Hi!", queue)
```

I added the `consume_arguments` to provide you the ability to pass an extra options to the `consume` method

**Reply by meirmedigate** [reply]
This is great!
There is also a native stream protocol, even if it is not added now, I think that it should be planned (it is supposed to be much more efficient).

**Comment by Lancetnik** [comment]
> This is great! There is also a native stream protocol, even if it is not added now, I think that it should be planned (it is supposed to be much more efficient).

Excuse me, but could you explain what the native stream protocol is? I don't seem to know him

If possible, attach any material on the topic, I will study it and add support to **Propan**

**Comment by Lancetnik** [comment]
Did you talk about [this feature](https://github.com/rabbitmq/rabbitmq-server/blob/v3.10.x/deps/rabbitmq_stream/docs/PROTOCOL.adoc)?

**Reply by meirmedigate** [reply]
Yes

**Reply by Lancetnik** [reply]
Well, let's wait for stable version

**Reply by meirmedigate** [reply]
Sounds like the right thing to do :-)
Just I would probably add it to the roadmap
